src/fhir/api.ts

```diff
@@ -128,3 +128,43 @@ const createEntry = (resource: Resource, identifier?: string) => {
     request: requestInfo,
   };
 };
+
+export const checkResponseOK = async (response: Response) => {
+  const r = await response.json();
+
+  if (!response.ok) {
+    console.error(r.body);
+    throw new Error(response.statusText);
+  }
+  if (!(r.type === "bundle-response")) {
+    return r;
+  }
+
+  const errors = (r as BundleResponse).entry
+    .filter((entry) => entry.response.status.toString().startsWith("4"))
+    .map((entry) => entry.response.outcome.issue);
+
+  if (errors.length > 1) {
+    errors.forEach((issue) => {
+      let message = "unknown error in bundle";
+      if (issue && issue.length > 0) {
+        message = issue[0].diagnostics;
+      }
+      throw new Error(message);
+    });
+  }
+
+  if (r.type === "batch-response") {
```

Comment:
I don't think this can ever be true, because we've already returned if the type isn't a batch response?
 `if (r.type === "batch-response") {`

Reply:
ah yes, will rejig